src/js/components/Select/SelectContainer.js

```diff
@@ -246,39 +251,66 @@ const SelectContainer = forwardRef(
         ) {
           nextActiveIndex += 1;
         }
-        if (nextActiveIndex !== options.length) {
+
+        const searchInput = searchRef.current;
+        const clearButton = clearRef.current;
+        if (
+          // Expect search input to immediately precede the clear button
+          ((shouldShowClearButton('top') &&
+            document.activeElement === searchInput) ||
+            // Expect clear button to immediately follow the last
+            // selectable option
+            (shouldShowClearButton('bottom') &&
+              nextActiveIndex === options.length)) &&
```

Comment:
I tried to not stray far from the design of having an active index of `-1` represent a special boundary position for the options collection. Here, `options.length` represents the opposite boundary position.

Whenever navigation needs to take place to or from boundary indices, the idea is to use the focussed element to inform the next keyboard navigation destination. I did this to avoid wrangling checks involving index offsets.